src/components/common/Directors.vue

```diff
@@ -591,75 +570,56 @@ export default class Directors extends Mixins(CommonMixin, DateMixin, DirectorMi
+    baseAddressEdit: ComponentIF[],
+    mailAddressEdit: ComponentIF[]
   }
 
-  /** Effective date for querying current directors and appointing/ceasing them. */
-  @Prop({ default: '' })
-  private asOfDate: string
-
-  /** Indicates whether this component should be enabled or not. */
-  @Prop({ default: true })
-  private componentEnabled: boolean
-
   // Local definitions of computed properties for static type checking.
   // NB: use non-null assertion operator to allow use before assignment
+  readonly isBComp!: boolean
   readonly entityIncNo!: string
-  readonly lastPreLoadFilingDate!: string
   readonly currentDate!: string
-  readonly currentFilingStatus!: FilingStatus
-  readonly isBComp!: boolean
+  readonly lastPreLoadFilingDate!: string
   readonly lastCODFilingDate!: string
   readonly lastAnnualReportDate!: string
   readonly entityFoundingDate!: string
 
-  // Local properties.
+  /** Indicates whether this component should be enabled or not. */
+  @Prop({ default: true })
+  readonly componentEnabled: boolean
+
+  /**
+   * Directors list from parent page, used to set our working directors list (allDirectors).
+   * As needed, we emit this back to parent (ie, update/sync).
+   */
+  @Prop({ default: () => [] })
+  readonly directors: DirectorIF[]
+
+  /** Effective date for fetching and appointing/ceasing directors. */
+  private asOfDate: string
+
+  /** Original directors from Legal API. */
+  private original: DirectorIF[] = []
+
+  /** Working directors data (also used as v-model for the HTML list). */
+  private allDirectors: DirectorIF[] = []
+
   private directorEditInProgress = false
-  private directors: Array<DirectorIF> = []
-  private directorsOriginal: Array<DirectorIF> = []
   private directorPreEdit = null // officer before edit
   private showNewDirectorForm = false
-  private draftDate: string = null
   private showPopup = false
   private activeIndex = -1
   private activeIndexCustomCease = -1
   private activeDirectorToDelete: DirectorIF = null
   private cessationDateTemp: string = null
   private isEditingDirector = false
   private messageIndex = -1
-  private director: DirectorIF = {
-    id: null,
-    officer: {
-      firstName: '',
-      lastName: '',
-      middleInitial: ''
-    },
-    deliveryAddress: {
-      streetAddress: '',
-      streetAddressAdditional: '',
-      addressCity: '',
-      addressRegion: '',
-      postalCode: '',
-      addressCountry: '',
-      deliveryInstructions: ''
-    },
-    mailingAddress: {
-      streetAddress: '',
-      streetAddressAdditional: '',
-      addressCity: '',
-      addressRegion: '',
-      postalCode: '',
-      addressCountry: '',
-      deliveryInstructions: ''
-    },
-    appointmentDate: null,
-    cessationDate: null,
-    cessationDateTemp: null,
-    actions: []
-  }
-  private inProgressAddress: AddressIF = null
+
+  /** V-model for new director form. */
+  private newDirector: DirectorIF = { ...EmptyDirector }
```

Comment:
This is close to the "copy constructor" concept. Maybe some day we can go to that.

Reply:
NB: this is a shallow copy but it's handled in EmptyDirector (see my other comment on this below).